test/adapter/json_api/errors_test.rb

```diff
@@ -0,0 +1,154 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    class Adapter
+      class JsonApi
+        class ErrorsTest < Minitest::Test
+
+          # see
+          # https://github.com/rails/rails/blob/4-2-stable/activemodel/lib/active_model/errors.rb
+          # The below allows you to do:
+          #
+          #   model = ModelWithErrors.new
+          #   model.validate!            # => ["cannot be nil"]
+          #   model.errors.full_messages # => ["name cannot be nil"]
+          class ModelWithErrors
+            # Required dependency for ActiveModel::Errors
+            extend ActiveModel::Naming
+            def initialize
+              @errors = ActiveModel::Errors.new(self)
+            end
+            attr_accessor :name
+            attr_reader   :errors
+
+            # The following methods are needed to be minimally implemented
+
+            def read_attribute_for_validation(attr)
+              send(attr)
+            end
+
+            def self.human_attribute_name(attr, options = {})
+              attr
+            end
+
+            def self.lookup_ancestors
+              [self]
+            end
+          end
+
+          def setup
+            @model = ModelWithErrors.new
+          end
+
+          class JsonApiErrorSerializer < ActiveModel::Serializer
+            attribute :source
+            attribute :detail
+
+            def source
+              {pointer: "data/attributes/#{object.first}"}
+            end
+
+            def detail
+              detail = object.last
+              if detail.respond_to?(:join)
```

Comment:
Basically, I'm just coding to think through what the implementation should look like, so there's cases like here where I'm not sure how I should be generating an error and details.  If 'detail' should always be a string, then I need to join a collection of details, but I can't easily convert to an array due to how ActiveModel::Error behaves etc. to the `join` just points out that I'm not sure what I'm dealing with :)

This PR is partly about generating discussion about how the implementation should look (in addition to actually writing it ;). 

So, that's why it's not tested :)